app/queries/customers_with_balance.rb

```diff
@@ -19,24 +18,23 @@ def query
       joins(left_join_complete_orders).
       group("customers.id").
       select("customers.*").
-      select("#{outstanding_balance_sum} AS balance_value").
-      select("#{balance_sum_time} AS balance_time")
+      select("#{outstanding_balance_sum} AS balance_value")
   end
 
   private
 
   attr_reader :enterprise
 
   def validate_arguments
-    return unless [enterprise, @customers].all?(&:blank?)
+    return unless [enterprise, @customers].all?(&:nil?)
```

Comment:
Noticed that an empty customer array could throw this exception. Fixed and added test